src/nvim/tui/tui.c

```diff
@@ -1575,7 +1576,7 @@ static void augment_terminfo(TUIData *data, const char *term,
   // can use colons like ISO 8613-6:1994/ITU T.416:1993 says.
   bool has_colon_rgb = false
     // per GNOME bug #685759 and bug #704449
-    || (vte_version >= 3600)
+    || (vte_version >= 3600 && !tmux && !screen)
     || iterm || iterm_pretending_xterm  // per analysis of VT100Terminal.m
     // per http://invisible-island.net/xterm/xterm.log.html#xterm_282
     || true_xterm;
```

Comment:
Ok, but that sounds like the logic is currently correct but it depends on the current configuration of previous checks. Why not make it explicit?

Regarding at least my first comment above: irrespective of `TERM`, _if_ we detect tmux/screen, then (correct me if wrong), _nothing else matter_. It doesn't matter if we also detected `iterm` or `iterm_pretending_xterm`; while running in tmux we _must not_ emit colon codes.